parrot/src/pfs_table.cc

```diff
@@ -1366,8 +1378,11 @@ int pfs_table::unlink( const char *n )
 {
 	pfs_name pname;
 	int result = -1;
+	char dirname[PFS_PATH_MAX];
+
+	path_dirname(n,dirname);
 
-	if(resolve_name(0,n,&pname,false)) {
+	if(resolve_name(0,dirname,&pname,X_OK|W_OK,false) && resolve_name(0,n,&pname,F_OK,false)) {
```

Comment:
Right, we need to check permissions on the parent if we change that directory entry rather than the item itself. That's done with the E_OK flag as of commit fe8825e13d31c38f64108cbb0bd0e8706df14a2b